Library/Homebrew/rubocops/dependency_order_cop.rb

```diff
@@ -58,9 +58,8 @@ def sort_conditional_dependencies!(ordered)
             ordered.each_with_index do |dep, pos|
               idx = pos+1
               match_nodes = build_with_dependency_name(dep)
-              idx1 = pos if match_nodes && !match_nodes.empty?
```

Comment:
`idx1` was supposed to be initialized as `nil` for each iteration, but due to this buggy code, it had value from previous iteration and that was causing the bug.